lib/TestReportHandler.rb

```diff
@@ -127,11 +127,23 @@ def pass
     return 
   end 
 
+#	Might need to alter ‘score’ calculation based on including/not including setup or teardown failures
+# Or, add the option to decide whether to include them
+
   def finalize
     report.status = 'completed'
     report.score = ((1 - @num_failures / @num_tests.to_f) * 100).round(2)
     report.result = (report.score == 100 ? 'pass' : 'fail')
     report.issued = DateTime.now
+    
+    num_passes = 0
+    report.test[0].action.each do |ac|
+      num_passes += 1 if ac.operation.result.code == 'pass' if ac.operation
```

Comment:
```suggestion
      num_passes += 1 if ac.operation&.result&.code == 'pass'
```
The ampersand (`&`) is the safe navigation operator. If it tries to send a method call to an object (in this case, sending the `result` method to `operation`) and the object is `nil`, then instead of raising an error, it just returns `nil`. In this way, we don't need to check `if ac.operation` -- because if it isn't the `&` is just going to return `nil` anyways. This is more stylistic/personal preference than anything else, so feel free to ignore.